keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -173,42 +175,33 @@ Since images can be pre-loaded, loaded outside the `kubelet` process, and
 garbage collected.. the list of images that required authentication in `kubelet`
 will not be a source of truth for how all images were pulled that are in the
 container runtime cache. To mitigate, images can be garbage collected at boot.
-And we will persist ensure metadata across reboot of host, and restart
+And for alpha, we will not persist ensure metadata across reboot of host, and restart
 of kubelet, and possibly look at a way to add ensure metadata for images loaded
 outside of kubelet. In beta we will add a switch to enable re-auth on boot for
 admins seeking that instead of having to garbage collect where they do not use
 or expect preloaded images since boot.
 
-
 ## Design Details
 
-Kubelet will track, in memory, a hash map for the credentials that were successfully used to pull an image. It has been decided that the hash map will be persisted to disk, in alpha.
+Kubelet will track, in memory, a hash map for the credentials that were successfully used to pull an image. The hash map
+will not be persisted to disk, in alpha. For alpha explicitly, we will not reuse or add other state manager concepts to kubelet.
 
-The persisted "cache" will undergo cleanup operations on a timely basis (by default once an hour).
+See PR linked above for detailed design / behavior documentation.
 
-The persistence of the on storage cache is mainly for restarting kubelet and/or node reboot.
+Kubelet will add a new flag, named `PullImageSecretRecheckDuration` to make
+the expired duration configurable. The default value could be 1d. For a pod with
+IfNotPresent image pull policy and an image pull secret, kubelet will recheck
+the secret after `PullImageSecretRecheckDuration`.
 
-The max size of the cache will scale with the number of unique cache entries * the number of unique images that have not been garbage collected. It is not expected that this will be a significant number of bytes. Will be verified by actual use in Alpha and subsequent metrics in Beta.
+To make the cluster in most secure situation, set `PullImageSecretRecheckDuration` to 0,
+which means always recheck.
 
-See `/var/lib/kubelet/image_manager_state` in [kubernetes/kubernetes#114847](https://github.com/kubernetes/kubernetes/pull/114847)
+If user doesn't want to do recheck, set `PullImageSecretRecheckDuration` to -1 to disable recheck.
 
-> ```
-> {
->   "images": {
->     "sha256:eb6cbbefef909d52f4b2b29f8972bbb6d86fc9dba6528e65aad4f119ce469f7a": {
->       "authHash": { ** per review comment use SHA256 here vs hash **
->         "115b8808c3e7f073": {
->           "ensured": true,
->           "dueDate": "2023-05-30T05:26:53.76740982+08:00"
->         }
->       },
->       "name": "daocloud.io/daocloud/dce-registry-tool:3.0.8"
->     }
->   }
-> }
-> ```
+For kubelet restart, recheck is acceptable, because kubelet only restart when upgrade or in maintennance modes in most cases.
```

Comment:
nit: 
```
Note: since the cache is not persisted to disk, a recheck will happen every kubelet restart. This is acceptable because kubelet only restarts during upgrades or in maintenance modes. In other words, it should be relatively infrequent (and much less frequent than the default value of `PullImageSecretRecheckDuration`)
```

Reply:
Done